### ISSUE TITLE
another ArrayIndexOutOfBoundsException in TypographyRule

### ISSUE THREAD

**Issue by danielnaber** [languagetool-org/languagetool#4335]
```
Feb 13 13:11:13 api24 languagetool-api[6487]: Caused by: java.lang.RuntimeException: Could not check sentence (language: Ukrainian): <sentcontent>–</sentcontent>
Feb 13 13:11:13 api24 languagetool-api[6487]: #011at org.languagetool.JLanguageTool$TextCheckCallable.getOtherRuleMatches(JLanguageTool.java:1829)
Feb 13 13:11:13 api24 languagetool-api[6487]: #011at org.languagetool.JLanguageTool$TextCheckCallable.call(JLanguageTool.java:1725)
Feb 13 13:11:13 api24 languagetool-api[6487]: #011at org.languagetool.JLanguageTool$TextCheckCallable.call(JLanguageTool.java:1695)
Feb 13 13:11:13 api24 languagetool-api[6487]: #011at org.languagetool.JLanguageTool.performCheck(JLanguageTool.java:1181)
Feb 13 13:11:13 api24 languagetool-api[6487]: #011... 10 more
Feb 13 13:11:13 api24 languagetool-api[6487]: Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 2 out of bounds for length 2
Feb 13 13:11:13 api24 languagetool-api[6487]: #011at org.languagetool.rules.uk.TypographyRule.match(TypographyRule.java:89)
Feb 13 13:11:13 api24 languagetool-api[6487]: #011at org.languagetool.JLanguageTool.checkAnalyzedSentence(JLanguageTool.java:1254)
Feb 13 13:11:13 api24 languagetool-api[6487]: #011at org.languagetool.JLanguageTool$TextCheckCallable.getOtherRuleMatches(JLanguageTool.java:1802)
Feb 13 13:11:13 api24 languagetool-api[6487]: #011... 13 more
```

**Comment by arysin** [comment]
I think this should be fixed in #4333 at least I can't reproduce this now.

**Comment by danielnaber** [comment]
I can reproduce with this URL: https://api.languagetool.org/v2/check?text=%E2%80%93&language=uk - your fix for #4333 is already online at languagetool.org, so any exception that occurs now seems to be something else.

**Comment by arysin** [comment]
Sorry I still can't easily reproduce it locally with just "\u2013" via unit test or command line, but I'll take a look deeper.
